### PR TITLE
Refactor Private.js file to Private.tsx file

### PR DESCRIPTION
**Pull request by DeividasK**
### Quick summary
I'm going to be adding a new route as part of OCR support work in operator UI. Refactoring `Private.js` file to typescript and functional component as part of an on-going migration to TS.

### CODE REVIEW

File: operator_ui/src/Private.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  return (
+    <Grid container spacing={0}>
+      <Grid item xs={12}>
+        <Header
+          onResize={(_width, height) => setHeaderHeight(height)}
+          drawerContainer={drawerContainer}
+        />
+        <main
+          ref={ref => {
+            drawerContainer = ref
```
Comment (hellobart) [inline operator_ui/src/Private.tsx:105]: Im not sure about the context of the functionality but shouldn't we use createRef/useRef here?

**Review by hellobart — DISMISSED** [review]

File: operator_ui/src/Private.tsx (hunk 1 above)
Comment (DeividasK) [reply operator_ui/src/Private.tsx:105]: Good shout, we should

File: operator_ui/src/containers/Jobs/Show.tsx [hunk 2]
```diff
@@ -206,16 +216,7 @@ export const Show: React.FC<Props> = ({
...
-  params: {
-    jobSpecId: string
-  }
-}
-
-const mapStateToProps = (
-  state: AppState,
-  ownProps: { match: Match; showJobRunsCount: number },
-) => {
+const mapStateToProps = (state: AppState, ownProps: ShowComponentProps) => {
```
[the remark this answers is not in the document]
Comment (DeividasK) [reply operator_ui/src/containers/Jobs/Show.tsx:219]: I think we shouldn't be using `mapStateToProps` at all and migrate to mapped `useSelector`, but I didn't want to change to many things at once to keep the size of the PR reasonable. Though I agree that we should refactor it once we get to the OCR work for the `Show` part.

File: operator_ui/src/Private.tsx (hunk 1 above)
Comment (DeividasK) [reply operator_ui/src/Private.tsx]: Actually, had to revert back, because `@material-ui` lib did not expect a ref object type.

**Review by hellobart — APPROVED** [review]

**Pull request merged**